packages/style-engine/class-wp-style-engine.php

```diff
@@ -319,31 +306,26 @@ public static function get_instance() {
 	 *
 	 * @param string $css_selector     When a selector is passed, the function will return a full CSS rule `$selector { ...rules }`, otherwise a concatenated string of properties and values.
 	 * @param array  $css_declarations An array of parsed CSS property => CSS value pairs.
-	 * @param string $store_key        A valid key corresponding to an existing store in static::$stores.
+	 * @param string $store_key        A valid store key.
 	 *
 	 * @return void.
 	 */
 	public static function store_css_rule( $css_selector, $css_declarations, $store_key ) {
-		if ( ! $css_selector || ! isset( static::$stores[ $store_key ] ) ) {
+		if ( empty( $css_selector ) || empty( $css_declarations ) ) {
 			return;
 		}
-		$css_declarations = new WP_Style_Engine_CSS_Declarations( $css_declarations );
-		$stored_css_rule  = static::$stores[ $store_key ]->add_rule( $css_selector );
-		$stored_css_rule->add_declarations( $css_declarations );
+		static::get_store( $store_key )->add_rule( $css_selector )->add_declarations( $css_declarations );
```

Comment:
Simplifies the implementation by using the methods other objects in the engine already have